src/Views/HotspotPage.vala

```diff
@@ -131,13 +133,12 @@
         }
 
         private void validate_entries () {
-            bool key_text_over_8 = key_entry.text.length >= 8;
-            status_switch.sensitive = key_text_over_8 || !sensitive;
-
-            if (!key_text_over_8 && key_entry.text != "") {
-                key_entry.set_icon_from_icon_name (Gtk.EntryIconPosition.SECONDARY, "process-error-symbolic");
-            } else {
-                key_entry.set_icon_from_icon_name (Gtk.EntryIconPosition.SECONDARY, null);
+            status_switch.sensitive = key_entry.is_valid || !sensitive;
+            if (!key_entry.is_valid) {
+                    key_entry.secondary_icon_tooltip_text = _("Password needs to be at least 8 characters long");
+                }
+            else {
+                key_entry.secondary_icon_tooltip_text = _("Password is at least 8 characters long");
```

Comment:
The indentation seems a little off for this section. We use 4 space indentation and cuddled else. You can find our code style guide here: https://docs.elementary.io/develop/writing-apps/code-style#vala
```suggestion
            if (!key_entry.is_valid) {
                key_entry.secondary_icon_tooltip_text = _("Password needs to be at least 8 characters long");
            } else {
                key_entry.secondary_icon_tooltip_text = _("Password is at least 8 characters long");
```